### PR TITLE
Integration of ODEs with complex dependent variables

### PR DESCRIPTION
**Pull request by lbenet**
This PR adds the possibility to integrate ODEs with complex dependent variables (the independent variable is still real), and includes some simple tests.

### CODE REVIEW

**Comment by lbenet** [comment]
Tests may not pass because some changes have been required in [TaylorSeries.jl](https://github.com/JuliaDiff/TaylorSeries.jl), which are already in master. I need to tag a new version in TaylorSeries in order to merge this, so please, hold me on.

I think tests do pass locally, but please check.

**Comment by PerezHz** [comment]
Thanks for this PR @lbenet ! I did check the tests for this PR locally, and they were successful! Using TaylorSeries latest master, output on my machine was:

```julia
WARNING: deprecated syntax "[a=>b for (a,b) in c]".
Use "Dict(a=>b for (a,b) in c)" instead.

WARNING: deprecated syntax "[a=>b for (a,b) in c]".
Use "Dict(a=>b for (a,b) in c)" instead.
Tests: dot{x}=x^2, x(0) = 1
WARNING: Maximum number of integration steps reached; exiting.
WARNING: Maximum number of integration steps reached; exiting.
11 facts verified.
Tests: dot{x}=x^2, x(0) = 3; nsteps <= maxsteps
42 facts verified.
Tests: dot{x}=x.^2, x(0) = [3.0,1.0]
WARNING: Maximum number of integration steps reached; exiting.
WARNING: Maximum number of integration steps reached; exiting.
14 facts verified.
Test non-autonomous ODE: dot{x}=cos(t)
18 facts verified.
Test integration of ODE with complex dependent variables
23 facts verified.
```

**Comment by lbenet** [comment]
In travis, the tests are not passing for different reasons, essentially because there is one missing method for `evaluate!` in TaylorSeries.jl (for Julia v0.4 and 0.5) and due to depreactions (in Julia v0.6).

At the moment I'm tagging a new version of TaylorSeries.jl, probably the final one supporting Julia 0.4; the idea is to tag a new one (which only supports Julia 0.5 and 0.6) which includes some deprecations for Julia 0.6. This should solve the problems for 0.6 (but creates new for Julia0.4); that new version should be the minimum required for TaylorIntegration.jl, though we would need to drop support for Julia 0.4.

So let's be patient a couple of days...

**Comment by lbenet** [comment]
The current tag for `TaylorSeries.jl` is 0.3.1, and this is the last version supporting Julia v0.4. 

I am about pushing a new commit, where I require 0.3.1 as the minimum version of `TaylorSeries.jl`, and noticed that we REQUIRE `Compat.jl`. It seems to me we do not use it, except for compatibility of `views`, i.e., for Julia v0.4 support. 

My point is that we should think about dropping Julia v0.4 support. Any thoughts @PerezHz ?

**Comment by PerezHz** [comment]
I agree @lbenet! We should focus on supporting Julia 0.5 and 0.6! So yeah perhaps it's best to "un-REQUIRE" Compat.jl and drop support for Julia 0.4.. That also would be consistent with what I understand you're currently doing with TaylorSeries.jl. Just went ahead with this in commit `c4141b0` and removed Compat.jl as well as the conditionals such as:

```julia
if VERSION < v"0.5-"
    @fact xT[1,1:end] --> x0'
else
    @fact xT[1,1:end] --> x0
end
```

which are no longer needed in `src/runtests.jl`

**Comment by PerezHz** [comment]
Incidentally, completed tests so far in travis and appveyor with Julia 0.4 are passing!

File: test/runtests.jl [hunk 1]
```diff
@@ -63,11 +63,6 @@ facts("Tests: dot{x}=x^2, x(0) = 3; nsteps <= maxsteps") do
     @fact length(tv) --> 14
     @fact length(xv[:,1]) --> 14
     @fact length(xv[:,2]) --> 14
-    if VERSION < v"0.5-"
-        @fact xv[1,1:end] --> q0'
-    else
-        @fact xv[1,1:end] --> q0
-    end
```
Comment (lbenet) [inline test/runtests.jl]: The line `@fact xv[1,1:end] --> q0` should remain there, i.e., where the Julia version is at least 0.5? The same on the other changes below in the file.

**Comment by lbenet** [comment]
Incidentally, travis is taking really too long to complete the tests!

File: test/runtests.jl (hunk 1 above)
Comment (lbenet) [inline test/runtests.jl]: Moreover: in `TaylorIntegration.jl` there is the line `import Compat.view` which should also be removed...

**Comment by PerezHz** [comment]
I think the tests on macOS (ie, OS X) are slowing travis...

File: test/runtests.jl (hunk 1 above)
Comment (PerezHz) [inline test/runtests.jl]: Sure! My mistake.. I'll correct this right away!

Comment (PerezHz) [inline test/runtests.jl]: Thanks!!

**Comment by PerezHz** [comment]
@lbenet just checked travis log file, it seems there was a bug in Julia 0.6 which was just fixed a couple hours ago in commit [03278a6](https://github.com/JuliaLang/julia/commit/03278a68f15e45e7f25bba90ca811922d7c79f5f), which essentially repeats endlessly a warning about Arrays. Travis is using an older commit (see TaylorIntegration travis [job #142.2 log file lines 156 and 157](https://travis-ci.org/PerezHz/TaylorIntegration.jl/jobs/200244703) ), and thus the tests run endlessly while displaying Array deprecation warnings.

Still, haven't figured why for [job #140.3](https://travis-ci.org/PerezHz/TaylorIntegration.jl/jobs/200215146) travis used a newer Julia commit, which caused the tests to finish successfully, although with a couple warnings (which will be fixed by TaylorSeries.jl latest release). If we can tell travis to use at least Julia 0.6 commit 03278a6, then tests will finish successfully... in a finite time.

**Comment by PerezHz** [comment]
Restarted the jobs which failed last time on travis, with a newer Julia 0.6 build; tests should pass now

**Comment by lbenet** [comment]
Finally, the tests are passing !

I noticed the problem you mentioned and tried to get rid of such occurrences, though it was done in a different PR in TaylorSeries.jl (see https://github.com/JuliaDiff/TaylorSeries.jl/pull/83), which is more advanced that v0.3.1. I think we should wait until I tag a new version in TaylorSeries.jl (0.4.0), without any support to Julia v0.4, and then include that version in the `REQUIRE` file here. This should happen in the next few days.

**Comment by lbenet** [comment]
Tests in linux (and windows) seem to pass; we should wait to get the tests also in Mac (though they pass locally in mine). Once this happens, this PR can be merged.

I would suggest to solve the deprecation warnings (in another PR) before tagging a new (minor) release. (A minor release, since we will not support Julia 0.4 from here on.) I will try to work on them tonight.

**Comment by lbenet** [comment]
Merging!

**Comment by PerezHz** [comment]
Great!! Thanks!! This is a really cool new feature!

**Pull request merged**